### ISSUE TITLE
Version 1.0

### ISSUE THREAD

**Issue by GenerallyIntelligent** [GenerallyIntelligent/hatespace#4]
To get a minimally functional package and test our seq2seq model we need to to the following:
- [ ] Clean up the analysis tools / integrate them with the rest of `hatespace`
- [ ] Get transformer embedder running (Add config changes to `decoder`)
- [ ] Implement `compute_loss` for the `HatspaceTrainer` class
- [ ] Implement the side information dataset (just for `IronMarch` dataset, for now)

**Comment by GenerallyIntelligent** [comment]
I'll work on getting the transformer embedder running. I'm the most familiar with the huggingface model structures, so that'll be easiest.

**Comment by GenerallyIntelligent** [comment]
The `HatespaceTrainer` class has been temporarily removed. The model's training procedure is likely to be more complicated than we first anticipated. It may be later until we package that functionality back into a trainer object.

**Comment by GenerallyIntelligent** [comment]
Additional tasks to complete:

- [ ] Fix bugs in generation code
- [ ] Test various inner embedder strategies

**Comment by HesitantlyHuman** [comment]
Tasks for getting the side information up and running Some of these don't directly apply to this repo, but [hatecomp](https://github.com/HesitantlyHuman/hatecomp). We'll put them here regardless so that everything is contained in one location:

- [ ] Train and upload final versions of the hatecomp classifiers
- [ ] Use the hatecomp classifiers to generate IronMarch side information
- [ ] Generate non-classifier side information (hate words, forum structure, etc)
- [ ] Update IronMarch dataset to support side information
- [ ] Create side loading function which will load side information from disk and add it to the IronMarch dataset
- [ ] Update the dataloader preparation functions to optionally call the new side loading function
- [ ] Add optional side information head to Model, and output predictions when the side information head is present
- [ ] Update the HatepaceTrainer to watch for side information predictions
- [ ] Update loss function to accept optional side information
- [ ] Add the side information to HatespaceTrainer's metric tracking
